action.yml

```diff
@@ -14,5 +14,17 @@ inputs:
 runs:
   using: 'composite'
   steps:
+    # Occasionally, there are rather large delays (> a few hours)
+    # between the time a pacakge is registered in General and
+    # propagated to pkg.julialang.org.  We can avoid this by manually
+    # cloning ~/.julia/registries/General/ in Julia 1.5 and later.
+    # See:
+    # * https://github.com/JuliaLang/Pkg.jl/issues/2011
+    # * https://github.com/JuliaRegistries/General/issues/16777
+    # * https://github.com/JuliaPackaging/PkgServer.jl/issues/60
+    - run: julia --color=yes -e 'using Pkg; VERSION >= v"1.5-" && !isdir(joinpath(DEPOT_PATH[1], "registries", "General")) && Pkg.Registry.add("General")'
+      shell: bash
+      env:
+        JULIA_PKG_SERVER: ""
```

Comment:
This only sets the variable for this particular step, is that intended? (I don't know how Pkg chooses where to get packages from)

Reply:
Yes. I tried to clarify why we are doing this in the comment:

https://github.com/julia-actions/julia-runtest/blob/ca99fb0712d4a4eaec4ebe3c78f639306e244e7c/action.yml#L27-L32

Reply:
I think this is intentional.

If the package is in the registry clones above, but not available in the package server, then Pkg will fall back to GitHub.

Reply:
Thanks for the explanation :)